cloud/src/autogluon/cloud/predictor/cloud_predictor.py

```diff
@@ -0,0 +1,861 @@
+        initial_instance_count: int, default = 1,
+            Initial number of instances to be deployed for the endpoint
+        wait: Bool, default = True,
+            Whether to wait for the endpoint to be deployed.
+            To be noticed, the function won't return immediately because there are some preparations needed prior deployment.
+        autogluon_sagemaker_inference_model_kwargs: dict, default = dict()
+            Any extra arguments needed to initialize AutoGluonSagemakerInferenceModel
+            Please refer to https://sagemaker.readthedocs.io/en/stable/api/inference/model.html#sagemaker.model.FrameworkModel for all options
+        **kwargs:
+            Any extra arguments needed to pass to deploy.
+            Please refer to https://sagemaker.readthedocs.io/en/stable/api/inference/model.html#sagemaker.model.Model.deploy for all options
+        """
+        if not predictor_path:
+            assert self.get_fit_job_status(self.fit_job_name) == 'Completed', 'Fit job not completed.'
+        if not predictor_path:
+            assert self.cloud_predictor_path, 'No cloud trained model found.'
+            predictor_path = self.cloud_predictor_path
+
+        if not endpoint_name:
+            endpoint_name = sagemaker.utils.unique_name_from_base("sagemaker-autogluon-serving-trained-model")
+        framework_version = self._parse_framework_version(framework_version, 'inference')
+
+        predictor_path = self._upload_predictor(predictor_path, f'endpoints/{endpoint_name}/predictor')
+        self.cloud_predictor_path = predictor_path
+
+        if self.predictor_type == 'tabular':
+            entry_point = tabular_serve_script
+        if self.predictor_type == 'text':
+            entry_point = text_serve_script
+        user_entry_point = autogluon_sagemaker_inference_model_kwargs.pop('entry_point', None)
+        if user_entry_point:
+            logger.warning(f'Providing a custom entry point could break the deployment. Please refer to `{entry_point}` for our implementation')
+            entry_point = user_entry_point
+
+        predictor_cls = AutoGluonRealtimePredictor
+        user_predictor_cls = autogluon_sagemaker_inference_model_kwargs.pop('predictor_cls', None)
+        if user_predictor_cls:
+            logger.warning('Providing a custom predictor_cls could break the deployment. Please refer to `AutoGluonRealtimePredictor` for how to provide a custom predictor')
+            predictor_cls = user_predictor_cls
+
+        model = AutoGluonSagemakerInferenceModel(
+            model_data=predictor_path,
+            role=self.role,
+            region=self.region,
+            framework_version=framework_version,
+            instance_type=instance_type,
+            entry_point=entry_point,
+            predictor_cls=predictor_cls,
+            **autogluon_sagemaker_inference_model_kwargs
+        )
+
+        logger.log(20, 'Deploying model to the endpoint')
+        self.endpoint = model.deploy(
```

Comment:
I think it's better to have only one endpoint for a cloud predictor. Otherwise the interface would be complicated. You'll need to pass in which endpoint you want to use while doing predict_realtime. Similar issue could happen with cleanup etc.